src/utils/resources.ts

```diff
@@ -35,3 +40,28 @@ export const referenceForObj = (obj: K8sResourceCommon) => {
   const { group, version, kind } = groupVersionKindForObj(obj);
   return referenceFor(group, version, kind);
 };
+
+/**
+ * Resolve GVK based on the provided list (or fallback if matching provider is missing).
+ */
+export const resolveProviderRef = (
+  { name, namespace }: ObjectReference,
+  providers: ProviderResource[],
+): ProviderRef => {
+  // fallback is required if provider reference inside plan is obsolete/out-of-synch
+  const fallbackProvider: Partial<ProviderResource> = {
+    apiVersion: CLUSTER_API_VERSION,
+    kind: 'Provider',
+  };
+
+  const provider = providers.find(
+    (p) => p.metadata?.namespace === namespace && p.metadata?.name === name,
+  );
+
+  return {
+    resolved: !!provider,
+    name,
+    gvk: groupVersionKindForObj(provider ?? fallbackProvider),
+    ready: hasCondition((provider?.status?.conditions ?? []) as IStatusCondition[], 'Ready'),
```

Comment:
i have looked at some providers, and is missing the new `phase` we suppose to use for 2.4, did you see it ? (using conditions should be deprecated in our code)

Reply:
OK. Can we fix it under separate PR? Then we could replace the `ready` columns in the Provider list screen and fix other places where we rely on conditions.